test/tools/isoltest.cpp

```diff
@@ -401,20 +401,25 @@ int main(int argc, char const *argv[])
 {
 	setupTerminal();
 
-	solidity::test::IsolTestOptions options(&TestTool::editor);
-
-	try
 	{
-		if (options.parse(argc, argv))
+		solidity::test::IsolTestOptions options(&TestTool::editor);
+
+		try
+		{
+			if (!options.parse(argc, argv))
+				return -1;
+
 			options.validate();
-		else
+			solidity::test::CommonOptions::setSingleton(std::move(options));
```

Comment:
I'm pretty sure this will slice ``options``. You can't just assign to a base class by value.

Reply:
I.e. you can't use an ``optional``, but you need a ``unique_ptr``.

Reply:
And a virtual destructor in ``CommonOptions``.

Reply:
You're right. A unique_ptr it is then